token-lending/program/src/processor.rs

```diff
@@ -970,6 +978,62 @@ fn _deposit_obligation_collateral<'a>(
     Ok(())
 }
 
+fn process_deposit_reserve_liquidity_and_obligation_collateral(
+    program_id: &Pubkey,
+    liquidity_amount: u64,
+    accounts: &[AccountInfo],
+) -> ProgramResult {
+    if liquidity_amount == 0 {
+        msg!("Liquidity amount provided cannot be zero");
+        return Err(LendingError::InvalidAmount.into());
+    }
+
+    let account_info_iter = &mut accounts.iter();
+    let source_liquidity_info = next_account_info(account_info_iter)?;
+    let user_collateral_info = next_account_info(account_info_iter)?;
+    let reserve_info = next_account_info(account_info_iter)?;
+    let reserve_liquidity_supply_info = next_account_info(account_info_iter)?;
+    let reserve_collateral_mint_info = next_account_info(account_info_iter)?;
+    let lending_market_info = next_account_info(account_info_iter)?;
+    let lending_market_authority_info = next_account_info(account_info_iter)?;
+    let destination_collateral_info = next_account_info(account_info_iter)?;
+    let obligation_info = next_account_info(account_info_iter)?;
+    let obligation_owner_info = next_account_info(account_info_iter)?;
+    let user_transfer_authority_info = next_account_info(account_info_iter)?;
+    let clock = &Clock::from_account_info(next_account_info(account_info_iter)?)?;
+    let token_program_id = next_account_info(account_info_iter)?;
+
+    let collateral_amount = _deposit_reserve_liquidity(
+        program_id,
+        liquidity_amount,
+        source_liquidity_info,
+        user_collateral_info,
+        reserve_info,
+        reserve_liquidity_supply_info,
+        reserve_collateral_mint_info,
+        lending_market_info,
+        lending_market_authority_info,
+        user_transfer_authority_info,
+        clock,
+        token_program_id,
+    )?;
+    _deposit_obligation_collateral(
```

Comment:
maybe there is a check on like collateral amount we should do?

Reply:
The _deposit_obligation_collateral checks for nonzero collateral amount. Anything else? I suupose we can change it to a >0 check

Reply:
just kidding its unsigned